File: archive/minimum-operations-to-make-the-array-increasing.js
const minOperations = function(nums) {
  if (nums.length < 2) {return 0}
  let prev = nums[0]
  let ans = 0
  for (let i = 1; i < nums.length; i++) {
    if (prev >= nums[i]) {
      let r = prev + 1 - nums[i]
      ans+=r
      nums[i]+=r
    }

    prev = nums[i]
  }

  return ans
};

console.log(minOperations([1,5,2,4,1]))

File: go-archive/minimum-operations-to-make-the-array-increasing.go
func minOperations(nums []int) int {
    ans:=0

    for i:=0; i < len(nums) - 1; i++ {
        if (nums[i + 1] <= nums[i]) {
            ans+=nums[i] - nums[i + 1] + 1
            nums[i + 1]+= nums[i] - nums[i + 1] + 1
        }
    }

    return ans
}